src/mpid/ch4/shm/posix/posix_init.c

```diff
@@ -299,7 +299,6 @@ int MPIDI_POSIX_comm_bootstrap(MPIR_Comm * comm)
             MPIR_ERR_CHECK(mpi_errno);
         }
 
-        int slab_size = MPIDI_POSIX_eager_shm_size(size);
```

Comment:
The size setting on line 271 includes extra space for `head_shm_size`. Is it needed here? The deleted line resized the slab down to not include the header.

Reply:
I think the extra space is need as the code on line 307 offset from the base of the slab with head_shm_size. Plus the deleted line would not work as `size` is not defined. The size here use to be the number of local processes.

Reply:
Ah, yes I didn't realize it would not even compile.